packages/select/src/vaadin-select.js

```diff
@@ -529,34 +591,55 @@ class Select extends DelegateFocusMixin(FieldMixin(SlotMixin(ElementMixin(Themab
 
     const label = selected.getAttribute('label');
     if (label) {
-      labelItem = this.__createItem(label);
+      labelItem = this.__createItemElement({ label });
     } else {
       labelItem = selected.cloneNode(true);
     }
 
     // store reference to the original item
     labelItem._sourceItem = selected;
 
-    this.__appendItem(labelItem);
+    this.__appendValueItemElement(labelItem);
 
     // ensure the item gets proper styles
     labelItem.selected = true;
   }
 
-  /** @private */
-  __createItem(text) {
-    const item = document.createElement('vaadin-item');
-    item.textContent = text;
-    return item;
+  /**
+   * @param {!SelectItem} item
+   * @private
+   */
+  __createItemElement(item) {
+    let itemElement;
+    if (item.component instanceof HTMLElement) {
+      itemElement = item.component;
```

Comment:
This (unpromoted/hidden and a bit hacky) feature was added to menu-bar just because it was required by the Flow Counterpart implementation. Since Select Flow doesn't have the same requirement, we probably shouldn't add the feature here.

Reply:
I'm not a fan of this feature too. Removed. 👍